net-p2p/gridcoin/gridcoin-5.3.3.0.ebuild

```diff
@@ -0,0 +1,164 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit flag-o-matic systemd desktop
+
+DESCRIPTION="Gridcoin Proof-of-Stake based crypto-currency that rewards BOINC computation"
+HOMEPAGE="https://gridcoin.us/"
+
+if [[ ${PV} == 9999 ]]; then
+		inherit git-r3
+		EGIT_REPO_URI="https://github.com/gridcoin/Gridcoin-Research.git"
+		EGIT_BRANCH="development"
+		PATCHES="${FILESDIR}/gridcoin-9999-desktop.patch"
+else
+		SRC_URI="https://github.com/gridcoin/Gridcoin-Research/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+		KEYWORDS="~amd64"
+		S="${WORKDIR}/Gridcoin-Research-${PV}"
+fi
+
+SLOT="0"
+LICENSE="MIT"
+
+IUSE_GUI="qt5 dbus"
+IUSE_DAEMON="daemon"
+IUSE_OPTIONAL="+boinc qrcode upnp pic libraries utils +harden bench ccache static debug test"
+IUSE="${IUSE_GUI} ${IUSE_DAEMON} ${IUSE_OPTIONAL}"
+
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="|| ( daemon qt5 ) dbus? ( qt5 ) qrcode? ( qt5 )"
+
+RDEPEND="
+	acct-group/gridcoin
+	acct-user/gridcoin
+	>=dev-libs/boost-1.73.0:=
+	dev-libs/libevent
+	>=dev-libs/libzip-1.3.0:=
+	>=dev-libs/openssl-1.1.1d:=
+	net-misc/curl
+	sys-libs/db:5.3[cxx]
+	boinc? ( sci-misc/boinc )
+	dbus? ( dev-qt/qtdbus:5 )
+	qrcode? ( media-gfx/qrencode )
+	qt5? ( dev-qt/qtcore:5 dev-qt/qtgui:5 dev-qt/qtnetwork:5 dev-qt/qtwidgets:5 dev-qt/qtconcurrent:5 dev-qt/qtcharts:5 )
+	upnp? ( >=net-libs/miniupnpc-1.9.20140401 )
+	utils? ( net-p2p/bitcoin-cli dev-util/bitcoin-tx )"
+DEPEND="${RDEPEND}
+	qt5? ( dev-qt/linguist-tools:5 )"
+
+pkg_setup() {
+	BDB_VER="$(best_version sys-libs/db:5.3)"
+	export BDB_CFLAGS="-I/usr/include/db${BDB_VER:12:3}"
+	export BDB_LIBS="-ldb_cxx-${BDB_VER:12:3}"
+
+}
+
+src_prepare() {
+	if use debug && [[ ! $(portageq envvar FEATURES) =~ .*(splitdebug|nostrip).* ]]; then
+		ewarn "You have enabled debug flags and macros during compilation."
+		ewarn "For these to be useful, you should also have Portage retain debug symbols."
+		ewarn "See https://wiki.gentoo.org/wiki/Debugging on configuring your environment"
+		ewarn "and set your desired FEATURES before (re-)building this package."
+	fi
+	default
+	./autogen.sh
```

Comment:
1. `|| die` on external commands
2. Don't call this anyway, use `autotools.eclass` (inherit it) and `eautoreconf`